quickstarts/microsoft.machinelearningservices/machine-learning-workspace-cmk/README.md

```diff
@@ -0,0 +1,192 @@
+
+## Prerequisites
+
+Before using this template, you must meet the following requirements:
+
+* The __Azure Machine Learning__ service principal must have __contributor__ access to your Azure subscription.
+* You must have an existing Azure Key Vault that contains an encryption key.
+* The Azure Key Vault must exist in the same Azure region where you will create the Azure Machine Learning workspace.
+* You must have an access policy in Azure Key Vault that grants __get__, __wrap__, and __unwrap__ access to the __Azure Cosmos DB__ application.
+
+To add the Azure Machine Learning service principal as a contributor to your subscription, you can use the following steps in the Azure CLI or Powershell for automation. Alternatively complete the steps via the Azure Portal.
+
+
+### Add Azure Machine Learning as a contributor
+
+1. Use the [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli) or [Azure Powershell](https://docs.microsoft.com/powershell/azure/install-az-ps) to authenticate to get your subscription ID:
+
+Azure CLI:
+
+```Bash
+    az account list --query '[].[name,id]' --output tsv
+```
+
+PowerShell:
+
+```powershell
+    Get-AzSubscription
+```
+
+From the list of subscriptions, select the one you want to use and copy the subscription ID and save the ID for the `Azure Machine Learning` entry.
+
+1. Get the object ID of the `Azure Machine Learning` service principal.
+
+Azure CLI:
+
+```Bash
+    az ad sp list --display-name "Azure Machine Learning" --query '[].[appDisplayName,objectId]' --output tsv
+```
+
+PowerShell:
+
+```powershell
+    Get-AzADServicePrincipal --DisplayName "Azure Machine Learning" | select-object DisplayName, Id
+```
+
+1. To add the service principal as a contributor to your subscription, use one of the following commands. Replace the `<subscription-ID>` with your subscription ID and `<object-ID>` with the ID for the service principal:
+
+Azure CLI:
+
+```Bash
+    az role assignment create --role 'Contributor' --assignee-object-id <object-ID> --subscription <subscription-ID>
+```
+
+PowerShell:
+
+```powershell
+    New-AzRoleAssignment --ObjectId <object-ID> --RoleDefinitionName "Contributor" -Scope /subscriptions/<subscription-ID>
+```
+
+### Add a key for encryption
+
+To generate a key in an existing Azure Key Vault, you can use the [prereq template](prereqs/prereq.azuredeploy.json) or complete the steps manually in the Azure Portal.
+
+### Enable customer-managed keys for Azure Cosmos DB
+
+See data encryption section of [Enterprise Security for Azure Machine Learning](https://docs.microsoft.com/azure/machine-learning/concept-enterprise-security#data-encryption) and [Configure customer-managed keys for your AzureCosmos account](https://docs.microsoft.com/azure/cosmos-db/how-to-setup-cmk).
+
+### Add an access policy to the key vault
+
+To add an access policy for Azure Cosmos DB to the key vault, use the following steps:
```

Comment:
where is cosmos used

Reply:
When creating a customer-managed key encryption workspace (CMK), Azure ML deploys a storage account, cosmos db resource and search resource on the customers' behalf in their subscription. For more details, see the readme and https://docs.microsoft.com/en-us/azure/machine-learning/concept-data-encryption#azure-cosmos-db.